Adding new GPS conversion convenience function.

enricgalceran opened the pull request.

ethzasl-jenkins commented:
Can one of the admins verify this patch?

tcies commented:
add to whitelist

tcies commented on msf_core/include/msf_core/gps_conversion.h:
```diff
@@ -22,19 +22,32 @@
 class GPSConversion {
- private:
-  msf_core::Quaternion ecef_ref_orientation_;
-  msf_core::Vector3 ecef_ref_point_;
```
+1 thanks for moving this

tcies commented on msf_core/include/msf_core/gps_conversion.h:
```diff
@@ -22,19 +22,32 @@
+
   GPSConversion();
-  void InitReference(const double& latitude, const double& longitude,
+  // default copy constructor and assignment operator are OK
```
Sentencify: Start with a cap, end with a period.

tcies commented on msf_core/include/msf_core/gps_conversion.h:
```diff
@@ -22,19 +22,32 @@
+
+  msf_core::Vector3 WGS84ToENU(const double& latitude,
+                               const double& longitude,
+                               const double& altitude) const; // convenience
```
Nit: No need for "convenience" comment.

tcies replied:
What? msf passes double by ref? This is inefficient. No need to refactor, but you could make your own function pass by copy.

tcies commented on msf_core/include/msf_core/gps_conversion.h:
```diff
@@ -22,19 +22,32 @@
+                               const double& longitude,
+                               const double& altitude) const; // convenience
+
+  void AdjustReference(const double& z_correction);
```
+1 for const, -1 for ref

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -21,14 +21,18 @@
 
-GPSConversion::GPSConversion() {
+GPSConversion::GPSConversion()
+{
```
This is not our convention. Please revert. Are you using the correct formatter? https://github.com/ethz-asl/programming_guidelines/wiki/Code-formatter

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -21,14 +21,18 @@
                                   const double & longitude,
-                                  const double & altitude) {
+                                  const double & altitude)
+{
```
same

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -51,9 +55,10 @@ void GPSConversion::InitReference(const double & latitude,
+msf_core::Vector3 GPSConversion::WGS84ToECEF(const double& latitude,
+                                             const double& longitude,
+                                             const double& altitude) const
+{
```
same

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -72,8 +77,8 @@ msf_core::Vector3 GPSConversion::WGS84ToECEF(const double & latitude,
-msf_core::Vector3 GPSConversion::ECEFToENU(
-    const msf_core::Vector3 & ecef) const {
+msf_core::Vector3 GPSConversion::ECEFToENU(const msf_core::Vector3& ecef) const
+{
```
same

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -81,10 +86,19 @@ msf_core::Vector3 GPSConversion::ECEFToENU(
+msf_core::Vector3 GPSConversion::WGS84ToENU(const double& latitude,
+                                            const double& longitude,
+                                            const double& altitude) const
+{
```
same

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -81,10 +86,19 @@ msf_core::Vector3 GPSConversion::ECEFToENU(
+}
+
+void GPSConversion::AdjustReference(const double& z_correction)
+{
```
same

tcies commented on msf_core/src/lib/gps_conversion.cc:
```diff
@@ -81,10 +86,19 @@ msf_core::Vector3 GPSConversion::ECEFToENU(
+                                            const double& longitude,
+                                            const double& altitude) const
+{
+    msf_core::Vector3 ecef = this->WGS84ToECEF(latitude, longitude, altitude);
```
const

tcies commented:
Some comments, otherwise looks good to me (scolgtm; = please address comments, but no need for a second review)

enricgalceran commented:
Thanks @tcies! Incorporated comments and refactored to pass POD types by reference on all functions in msf_core/gps_conversion.

Merging onto master.

markusachtelik commented on msf_core/include/msf_core/gps_conversion.h:
```diff
@@ -22,19 +22,34 @@
-                                const double& altitude) const;
+  // Default copy constructor and assignment operator are OK.
+
+  ~GPSConversion();
```
can you please reformat class / method names according to this:
https://github.com/ethz-asl/programming_guidelines/wiki/Cpp-Coding-Style-Guidelines#10-abbreviations-and-acronyms-must-not-be-uppercase-when-used-as-name-4

I think it's still not too late ;)
